dataset_doi.yaml

```diff
@@ -6,7 +6,7 @@ eia176:
   sandbox_doi: 10.5072/zenodo.3158
 eia860:
   production_doi: 10.5281/zenodo.4127026
-  sandbox_doi: 10.5072/zenodo.3299
```

Comment:
These changed because I needed to re-run test datasets with `--initialize` - before the draft discard change in this PR, updating an existing sandbox dataset would get Zenodo into a weird state. Anytime after you updated an existing dataset, a request for `/records/<concept_record_id>` would return `410: Resource deleted`.